### DISCUSSION TITLE
AES GCM Decryption

### DISCUSSION THREAD

**Discussion by gampam2000** [micropython/micropython#11952] [Libraries & Drivers] [answered]
Hi,
i need for a project the decryption of AES GCM with a 128bit key  and 96bit IV.
Micropython cryptolib seems not to support AES-GCM. 
So far my google research did not found any micropython implementations for it.
Does someone archived AES-GCM decryption?

Thanks & best regards

**Comment by dmazzella** [comment, marked as answer]
Hi,
try [ucryptography](https://github.com/dmazzella/ucryptography)

in [tests/ucryptography/ciphers.py](https://github.com/dmazzella/ucryptography/blob/master/tests/ucryptography/ciphers.py) there is an example of usage

Regards,
D.

**Reply by gampam2000** [reply]
Hi,
thanks for this, looks promising. I will try it.

Thanks & best regards
